Extract out DownloadManager from download service.

roeierez opened the pull request:
This PR separates (extract out) a small "DownloadManager" interface from the currently MobileDownloadService.
This change allows embedding anytime inside other apps and use the same FlutterDownloader plugin, overcoming its singleton requirement.
In addition it enables using a different implementation of download service and tasks management.
This PR should not cause in any behavioural changes.

amugofjava commented:
Hi Roei. PR looks great. Just one minor thing: I always use package style imports, so to keep things consistent can you change the `downloadable.dart` import to a package style rather than a relative style. Thanks.

roeierez commented:
> Hi Roei. PR looks great. Just one minor thing: I always use package style imports, so to keep things consistent can you change the `downloadable.dart` import to a package style rather than a relative style. Thanks.

Thanks Ben for your fast review. I actually see that I imported the file twice. Removed the extra line and squashed.
